drivers/misc/rwbuffer.c

```diff
@@ -140,18 +140,28 @@ static void rwb_wrflush(FAR struct rwbuffer_s *rwb)
   if (rwb->wrnblocks > 0)
     {
       size_t padblocks;
+      size_t wrnblocks;
 
       finfo("Flushing: blockstart=0x%08lx nblocks=%d from buffer=%p\n",
             (long)rwb->wrblockstart, rwb->wrnblocks, rwb->wrbuffer);
 
       padblocks = rwb->wrblockstart % rwb->wralignblocks;
       if (padblocks)
         {
+          wrnblocks = (rwb->wralignblocks - padblocks) < rwb->wrnblocks ?
+                (rwb->wralignblocks - padblocks) : rwb->wrnblocks;
           memmove(rwb->wrbuffer + padblocks * rwb->blocksize,
-                  rwb->wrbuffer, rwb->wrnblocks * rwb->blocksize);
+                  rwb->wrbuffer, wrnblocks * rwb->blocksize);
```

Comment:
but, memmove with partial block will destroy the content at the end, @ChenChuangForLinux .

Reply:
The padblocks + wrnblocks <= wralignblocks, why will the memmove destroy the content at the end?

Reply:
Since wrnblocks may smaller than rwb->wrnblocks, [rwb->wrnblocks - wrnblocks, rwb->wrnblocks) is destroyed by memmove.

Reply:
Oh, you're right, it my fault. I will take a new PR in 3 hours. By the way, I don't think it't necessarily to use padblocks strategy in rwb_wrflush(), just call rwb->wrflush() is simple, since ftl layer has done enough precoess.

Reply:
After more thinking, it's hard to handle unalignment issue here when the cache is near full without additional buffer. So, I think the better approach may allocate the buffer with one additional aligned block, or ensure the startblock is aligned at the first place.

Reply:
Sorry, I'm busy today. Why not remove all unalignment issue handler in rwb_wrflush()? ftl layler has been done it.

Reply:
No, ftl depends on the rwb alignment at all.

Reply:
But in another case in rwb_writebuffer()(rwbuffer.c:367), startblock may be unalignment, it also just call rwb->wrflush(ftl_flush).

Reply:
This my final solution based on the original idea.